sklearn/utils/estimator_checks.py

```diff
@@ -210,6 +217,26 @@ def _yield_clustering_checks(name, clusterer):
     yield check_non_transformer_estimators_n_iter
 
 
+def _yield_outliers_checks(name, estimator):
+
+    # checks for all outlier detectors
+    yield check_outliers_fit_predict
+
+    # checks for estimators that can be used on a test set
+    if name not in NO_TEST_SET_DETECTORS:
```

Comment:
Can't you just use hasattr here? I.e. you want to test that *if* it has predict, it also has the other behaviours